NoSuchMethodError with openapi generator since 1.12.0

Hello,
with the update from 1.11.1 to 1.12.0 the openapiApiGenerate task from the openapi plugin runs into a NoSuchMethodError:

```
Caused by: java.lang.NoSuchMethodError: 'com.fasterxml.jackson.databind.ObjectMapper io.swagger.v3.core.util.ObjectMapperFactory.createYaml31()'
	at io.swagger.v3.core.util.Yaml31.mapper(Yaml31.java:20)
	at io.swagger.v3.parser.util.DeserializationUtils.<clinit>(DeserializationUtils.java:126)
	at io.swagger.v3.parser.OpenAPIV3Parser.readContents(OpenAPIV3Parser.java:168)
	at io.swagger.v3.parser.OpenAPIV3Parser.readLocation(OpenAPIV3Parser.java:97)
	at io.swagger.parser.OpenAPIParser.readLocation(OpenAPIParser.java:16)
	at org.openapitools.codegen.config.CodegenConfigurator.toContext(CodegenConfigurator.java:653)
	at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:711)
```

Looks like the plugin takes the wrong swagger.v3.core version.

A minimal build.gradle.kts:

```
plugins {
    kotlin("jvm") version "1.9.21"
    id("com.github.imflog.kafka-schema-registry-gradle-plugin") version "1.12.0"
    id("org.openapi.generator") version "7.1.0"

}

group = "org.example"
version = "1.0-SNAPSHOT"

repositories {
    mavenCentral()
}

dependencies {
    testImplementation("org.jetbrains.kotlin:kotlin-test")
}

tasks.test {
    useJUnitPlatform()
}
kotlin {
    jvmToolchain(17)
}

openApiGenerate {
    generatorName.set("kotlin-spring")
    inputSpec.set("$rootDir/specs/test-api-v1.0.yaml")
    outputDir.set(layout.buildDirectory.dir("generated").get().asFile.absolutePath)
    configFile.set("$rootDir/src/main/resources/api-config.yaml")
    templateDir.set("$rootDir/src/main/resources/openapigen-templates")
    configOptions.set(mapOf("useSpringBoot3" to "true"))
}

```

Does anyone has an advice what to do?

With version 1.11.1 and the additional buildscript, everything works.

--- a/README.md
+++ b/README.md
@@ -9,10 +9,15 @@ for Gradle builds.
 
 ## Usage
 
-Follow the [gradle plugin portal instructions](https://plugins.gradle.org/plugin/com.github.imflog.kafka-schema-registry-gradle-plugin).
+The plugin relies on two libraries that are not available on common repositories, you will have to add them to
+your `buildscript`:
 
-<details>
-<summary>For plugin version < 1.12.0</summary>
+* one developed by confluent for which you need to add the `https://packages.confluent.io/maven/`
+* one developed by myself to add new capability to Avro for which you need to add the `https://jitpack.io`
+
+<div class="tabbed-code-block">
+  <details open>
+    <summary>Groovy</summary>
 
 ```groovy
 buildscript {
@@ -30,7 +35,30 @@ plugins {
     id "com.github.imflog.kafka-schema-registry-gradle-plugin" version "X.X.X"
 }
 ```
-</details>
+
+  </details>
+  <details>
+    <summary>Kotlin</summary>
+
+```kotlin
+buildscript {
+    repositories {
+        gradlePluginPortal()
+        maven("https://packages.confluent.io/maven/")
+        maven("https://jitpack.io")
+    }
+}
+plugins {
+    id("com.github.imflog.kafka-schema-registry-gradle-plugin") version "X.X.X"
+}
+```
+
+  </details>
+</div>
+
+Where "X.X.X" is the current version,
+see [gradle plugin portal](https://plugins.gradle.org/plugin/com.github.imflog.kafka-schema-registry-gradle-plugin) for
+details.
 
 ## Tasks
 
@@ -60,8 +88,9 @@ schemaRegistry {
   Could be removed if https://github.com/gradle/gradle/issues/1010 is fixed.
 * `outputDirectory` is the directory where action result will be stored as files (only register for now).
   This is an optional parameter.
-* `pretty` is whether the downloaded Avro or json schemas should be formatted ("pretty-printed") or minified. 
+* `pretty` is whether the downloaded Avro or json schemas should be formatted ("pretty-printed") or minified.
   This is an optional parameter.
+
 ### Download schemas
 
 Like the name of the task imply, this task is responsible for retrieving schemas from a schema registry.
@@ -322,7 +351,7 @@ here: [org.apache.kafka.common.config.SslConfigs](https://github.com/confluentin
 
 ### Examples
 
-Detailed examples can be found in the [examples directory](examples).
+Detailed examples can be found in the [examples' directory](examples).
 
 ## Version compatibility
 
@@ -333,7 +362,7 @@ It was easier to introduce all the changes in one shot instead of supporting bot
 users:
 
 * plugin versions above 1.X.X support the confluent version > 5.5.X (Avro / Json / Protobuf)
-* older plugin versions should support anything below 5.4.X
+* plugin versions should support anything below 5.4.X
 
 ## Developing
 
@@ -372,7 +401,10 @@ buildscript {
         // The new repository to import, you may not want this in your final gradle configuration.
         mavenLocal()
         maven {
-            url "http://packages.confluent.io/maven/"
+            url "https://packages.confluent.io/maven/"
+        }
+        maven {
+          url = "https://jitpack.io"
         }
     }
     dependencies {

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -7,7 +7,6 @@ version = "1.13.1-SNAPSHOT"
 plugins {
     kotlin("jvm") version "1.9.22"
     id("com.gradle.plugin-publish") version "1.2.1"
-    id("com.github.johnrengelman.shadow") version "7.1.2"
 }
 
 repositories {
@@ -20,13 +19,10 @@ repositories {
 val confluentVersion = "7.6.0"
 val avroVersion = "1.11.2"
 dependencies {
-    shadow(gradleApi())
-    implementation(platform("io.confluent:kafka-schema-registry-parent:$confluentVersion"))
-    implementation("io.confluent", "kafka-schema-registry") {
+    implementation(gradleApi())
+    implementation("io.confluent", "kafka-schema-registry", confluentVersion) {
         exclude("org.slf4j", "slf4j-log4j12")
     }
-    implementation("io.confluent", "kafka-json-schema-provider")
-    implementation("io.confluent", "kafka-protobuf-provider")
     // Our custom avro version. See https://github.com/ImFlog/avro
     implementation("com.github.ImFlog.avro", "avro", avroVersion)
 }
@@ -38,6 +34,10 @@ tasks.withType<KotlinCompile>().configureEach {
             "-Xself-upper-bound-inference"
         )
     }
+}
+
+java {
+    withSourcesJar()
 }
 
 // Unit tests
@@ -90,14 +90,6 @@ task<Test>("integrationTest") {
 }
 
 // Publish plugin
-tasks {
-    shadowJar {
-        archiveClassifier.set("")
-        from(sourceSets.main.get().output)
-        from(project.configurations.runtimeClasspath)
-    }
-}
-
 val registryPluginName = "com.github.imflog.kafka-schema-registry-gradle-plugin"
 @Suppress("UnstableApiUsage")
 gradlePlugin {

--- a/examples/build.gradle.kts
+++ b/examples/build.gradle.kts
@@ -1,7 +1,9 @@
 buildscript {
     repositories {
         gradlePluginPortal()
-//        mavenLocal()
+        mavenLocal()
+        maven("https://packages.confluent.io/maven/")
+        maven("https://jitpack.io")
     }
 
     dependencies {

